qutebrowser/mainwindow/statusbar/command.py

```diff
@@ -134,15 +135,25 @@ def set_cmd_text_command(self, text: str,
             count: The count if given.
             space: If given, a space is added to the end.
             append: If given, the text is appended to the current text.
+            insert: If given, the text is inserted in the current text
+                    at cursor position. This cannot be used with append
             run_on_count: If given with a count, the command is run with the
                           given count rather than setting the command text.
         """
         if space:
             text += ' '
+        if append and insert:
+            raise cmdutils.CommandError(
+                "Can't insert and append at the same time")
         if append:
             if not self.text():
                 raise cmdutils.CommandError("No current text!")
             text = self.text() + text
+        elif insert:
+            old_cursor_position = self.cursorPosition()
+            new_cursor_position = old_cursor_position + len(text)
+            text = (self.text()[:old_cursor_position] +
+                    text + self.text()[old_cursor_position:])
```

Comment:
Instead of fiddling with the text and cursor position by hand, you can just do `self.insert(text)`, as `Command` inherits from `misc.CommandLineEdit` which is a `QLineEdit` ([docs](https://doc.qt.io/qt-5/qlineedit.html#insert))

Reply:
I was considering it at first, but wasn't willing to interrupt the method natural continuation afterward (run_on_count, set_cmd_text, and possibly the STARTCHARS check).
Will do